### PR TITLE
Final updates for merge

### PR DESCRIPTION
**Pull request by BernieWhite**
# Overview/Summary

Clean ups for merging to `main`.

Related to #206

## This PR fixes/adds/changes/removes

1. Added additional comments and references.
2. Added suppression group for bastion public IP.
3. Added tag on Bastion public IP for suppression group and later rule exclusion.

## Testing Evidence

Tested locally with PSRule.

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated tests *(if required)* [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml) - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows) - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by BernieWhite** [comment]
@jtracey93 This should be good to go. Let me know if there is any further updates required.

**Review by jtracey93 — DISMISSED** [review]

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep [hunk 1]
```diff
@@ -258,7 +258,9 @@ module modBastionPublicIP '../publicIp/publicIp.bicep' = if (parBastionEnabled)
       publicIPAddressVersion: 'IPv4'
       publicIPAllocationMethod: 'Static'
     }
-    parTags: parTags
+    parTags: union({
+      'resource-usage': 'azure-bastion'
```
[the remark this answers is not in the document]
Comment (BernieWhite) [reply infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:262]: @jtracey93 Currently Azure uses various tags such as `ms-resource-usage` on cloud shell storage to indicate specifically what the storage account is for. Previously it made sense to use a similar tag to identify how a generic resource is being used, irrespective of PSRule. Additionally there is cases where Azure is `hidden:` tags which are not visible in the portal, but often used to link resources together.

The `Azure.PublicIP.AvailabilityZone` now in preview excludes `resource-usage` = `azure-bastion` so a suppression will not be required when v1.16.0 releases.

Separately you could suppress based on any criteria that identifies the resource as a Bastion resource. Which there only seems to be either tags or name that have any relevance in the ALZ-Bicep code. While you could use `name` is the customer configures the name they would also need to change the suppression which may be confusing so I would suggest that tag is the best option, but happy to take alternatives if you have any different ideas.

[the remark this answers is not in the document]
Comment (jtracey93) [reply infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:262]: Thanks for the explanation @BernieWhite. 

When is `v1.16.0` due to land?

My thinking is that as these tests are only for us on the `alz-bicep` repo to show consumers the code quality I dont want to influence/enforce tags upon on consumers as we have heard in ALZ community that they very much want the control of the tags.

I think maybe based on name is the best way forward here, agreed?

[the remark this answers is not in the document]
Comment (BernieWhite) [reply infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep]: @jtracey93 Updated based on name as discussed. PSRule for Azure is currently at v1.18.1.

**Comment by BernieWhite** [comment]
@jtracey93 Ready for final review and merge.

**Comment by jtracey93** [comment]
Thanks @BernieWhite

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**